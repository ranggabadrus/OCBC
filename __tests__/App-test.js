import React, {useEffect, useState} from 'react';
import AsyncStorage from '@react-native-async-storage/async-storage';

import Login from '../src/Components/Login';
import {
  render,
  fireEvent,
  waitFor,
  act,
  cleanup,
} from '@testing-library/react-native';
import Register from '../src/Components/Register';
import {useNavigation} from '@react-navigation/native';
import Dashboard from '../src/Pages/Dashboard';

jest.useFakeTimers();
// jest.runAllTimers();
jest.setTimeout(30000);

jest.mock('@react-navigation/native', () => {
  return {
    useNavigation: jest.fn(),
  };
});
afterEach(() => cleanup());

describe('Transaction Page', () => {});

describe('Dashboard Page ', () => {
  beforeEach(() => {
    fetch.mockResponses(
      [
        JSON.stringify({
          accountNo: '12345',
          balance: 111111,
          status: 'success',
        }),
      ],
      [
        JSON.stringify({
          status: 'success',
          data: [
            {
              transactionId: '622ba782f3aa395d71354900',
              amount: 100000,
              transactionDate: '2022-03-11T19:48:18.249Z',
              description: 'Default money-in transaction',
              transactionType: 'received',
              sender: {
                accountNo: '1111-111-1111',
                accountHolder: 'Jane',
              },
            },
          ],
        }),
      ],
    );
  });
  it('render dashboard with fetched data', async () => {
    const navigation = jest.fn();
    useNavigation.mockImplementation(() => ({
      navigate: navigation,
      addListener: navigation,
      replace: navigation,
    }));

    const {toJSON, getByTestId, findByTestId, findAllByTestId} = render(
      <Dashboard />,
    );

    await waitFor(async () => {
      const balance = getByTestId('balance', {}, {timeout: 2000});
      expect(balance.props.children).not.toEqual('NaN');
      const accountNo = getByTestId('accountNo', {}, {timeout: 2000});
      expect(accountNo.props.children).toEqual('12345');
      const transaction = getByTestId('transaction', {}, {timeout: 2000});
      expect(transaction.props.data).toHaveLength(1);
    });

    // expect(toJSON()).toMatchSnapshot();
  });

  it('from dashboard go to transfer page', async () => {
    const navigation = jest.fn();
    useNavigation.mockImplementation(() => ({
      navigate: navigation,
      addListener: navigation,
      replace: navigation,
    }));

    const {toJSON, getByTestId, findByTestId, findAllByTestId} = render(
      <Dashboard />,
    );

    await waitFor(async () => {
      const transfer = getByTestId('goToTransfer');
      fireEvent.press(transfer);
      expect(navigation).toHaveBeenCalledWith('Transfer');
    });

    expect(toJSON()).toMatchSnapshot();
  });

  it('open modal for detail transaction', async () => {
    const navigation = jest.fn();
    useNavigation.mockImplementation(() => ({
      navigate: navigation,
      addListener: navigation,
      replace: navigation,
    }));

    const {toJSON, getByTestId, findByTestId, findAllByTestId} = render(
      <Dashboard />,
    );

    await waitFor(async () => {
      const button = getByTestId('detail');
      fireEvent.press(button);
      const detailModal = getByTestId('detailModal');
      expect(detailModal.props.visible).toEqual(true);
    });

    expect(toJSON()).toMatchSnapshot();
  });
});

describe('Login page', () => {
  test('Validation text appear when fields are empty ', async () => {
    const {getByTestId, getByText, queryByTestId, toJSON, queryAllByTestId} =
      render(<Login />);

    const button = getByTestId('login');
    fireEvent.press(button);

    await waitFor(() => {
      expect(queryByTestId('emptyUsername')).toBeTruthy();
      expect(queryByTestId('emptyPassword')).toBeTruthy();
    });
  });

  test('Go to dashboard page and save token if credential corrects', async () => {
    const navigation = jest.fn();
    useNavigation.mockImplementation(() => ({replace: navigation}));

    const {findByText, getByTestId, debug, queryByText, toJSON, queryByTestId} =
      render(<Login />);

    beforeEach(() => {
      fetch.resetMocks();
    });

    fetch.mockResponseOnce(
      JSON.stringify({
        status: 'success',
        token: '12345',
        username: 'rangga',
      }),
    );

    const username = getByTestId('username');
    fireEvent.changeText(username, 'rangga');

    const password = getByTestId('password');
    fireEvent.changeText(password, 'rangga');

    const button = getByTestId('login');
    fireEvent.press(button);

    fetch('https://green-thumb-64168.uc.r.appspot.com/login').then(res => {
      expect(res.username).toEqual('rangga');
    });

    await waitFor(() => {
      expect(AsyncStorage.setItem).toBeCalledWith('token', '12345');
      expect(AsyncStorage.setItem).toBeCalledWith('username', 'rangga');
      expect(navigation).toHaveBeenCalledWith('Dashboard');
    });
  });
});

describe('Register page', () => {
  test('Validation text appear when fields are empty ', async () => {
    const {getByTestId, getByText, queryByTestId, toJSON} = render(
      <Register />,
    );

    const button = getByTestId('register');
    fireEvent.press(button);

    await waitFor(() => {
      expect(queryByTestId('emptyRegisterUsername')).toBeTruthy();
      expect(queryByTestId('emptyRegisterPassword')).toBeTruthy();
      expect(queryByTestId('emptyRegisterConfirmPassword')).toBeTruthy();
    });
  });

  test('Go to dashboard page and save token if register success', async () => {
    const navigation = jest.fn();
    useNavigation.mockImplementation(() => ({replace: navigation}));

    const {findByText, getByTestId, debug, queryByText, toJSON, queryByTestId} =
      render(<Register />);

    beforeEach(() => {
      fetch.resetMocks();
    });

    fetch.mockResponseOnce(
      JSON.stringify({
        status: 'success',
        token: '12345',
        username: 'rangga',
      }),
    );

    const username = getByTestId('registerUsername');
    fireEvent.changeText(username, 'rangga');

    const password = getByTestId('registerPassword');
    fireEvent.changeText(password, 'rangga');

    const confirmPassword = getByTestId('registerConfirmPassword');
    fireEvent.changeText(confirmPassword, 'rangga');

    const button = getByTestId('register');
    fireEvent.press(button);

    fetch('https://green-thumb-64168.uc.r.appspot.com/register').then(res => {
      expect(res.username).toEqual('rangga');
    });

    await waitFor(() => {
      expect(AsyncStorage.setItem).toBeCalledWith('token', '12345');
      expect(AsyncStorage.setItem).toBeCalledWith('username', 'rangga');
      expect(navigation).toHaveBeenCalledWith('Dashboard');
    });
  });
});
