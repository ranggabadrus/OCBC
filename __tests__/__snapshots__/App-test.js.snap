// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Page  from dashboard go to transfer page 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "position": "absolute",
        "top": 0,
      }
    }
  >
    <RNSVGSvgView
      bbHeight={500}
      bbWidth={750}
      focusable={false}
      height={500}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 500,
            "width": 750,
          },
        ]
      }
      width={750}
    >
      <RNSVGGroup>
        <RNSVGCircle
          cx={-50}
          cy={-300}
          fill={4278190208}
          propList={
            Array [
              "fill",
              "stroke",
              "strokeWidth",
            ]
          }
          r={500}
          stroke={4278190208}
          strokeWidth="2"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    collapsable={false}
    style={
      Object {
        "alignSelf": "flex-end",
        "backgroundColor": "#f2f2f2",
        "margin": 10,
        "transform": Array [
          Object {
            "translateX": 55,
          },
        ],
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "gray",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ó°—½
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginLeft": 5,
            "opacity": 1,
          }
        }
      >
        <Text>
          Logout
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
          "backgroundColor": "#fff",
          "borderBottomEndRadius": 50,
          "borderColor": "navy",
          "borderTopEndRadius": 50,
          "borderWidth": 2,
          "elevation": 20,
          "opacity": 0.9,
          "padding": 20,
          "width": 600,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
      testID="money"
    >
      You have
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 30,
            "fontWeight": "bold",
          },
        ]
      }
    >
      SGD$
      <Text
        testID="balance"
      >
        111,111.00
      </Text>
    </Text>
    <Text>
      Account No
    </Text>
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
      testID="accountNo"
    >
      12345
    </Text>
    <Text>
      Account Holder
    </Text>
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 20,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 20,
            "fontWeight": "bold",
          },
          Object {
            "marginBottom": 10,
          },
        ]
      }
    >
      Your transaction history
    </Text>
    <RCTScrollView
      contentContainerStyle={
        Object {
          "paddingBottom": 50,
        }
      }
      data={
        Array [
          Object {
            "amount": 100000,
            "description": "Default money-in transaction",
            "sender": Object {
              "accountHolder": "Jane",
              "accountNo": "1111-111-1111",
            },
            "transactionDate": "2022-03-11T19:48:18.249Z",
            "transactionId": "622ba782f3aa395d71354900",
            "transactionType": "received",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      testID="transaction"
      viewabilityConfigCallbackPairs={Array []}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="detail"
          >
            <View
              style={
                Object {
                  "backgroundColor": "#fff",
                  "borderColor": "#eaeaea",
                  "borderRadius": 20,
                  "borderWidth": 2,
                  "height": 100,
                  "marginVertical": 5,
                  "padding": 10,
                }
              }
            >
              <Text>
                3/12/2022, 2:48:18 AM
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "flexDirection": "row",
                    },
                    Object {
                      "justifyContent": "space-between",
                    },
                  ]
                }
              >
                <View>
                  <Text>
                    Jane
                  </Text>
                  <Text>
                    1111-111-1111
                  </Text>
                </View>
                <View>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#20cb9d",
                        },
                        Object {
                          "fontWeight": "bold",
                        },
                      ]
                    }
                  >
                    +SGD$
                    100,000.00
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 0,
        "left": 0,
        "marginBottom": 20,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "black",
          "borderRadius": 50,
          "opacity": 1,
          "paddingHorizontal": 50,
          "paddingVertical": 10,
        }
      }
      testID="goToTransfer"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontSize": 20,
              "fontWeight": "bold",
            },
            Object {
              "color": "#fff",
            },
          ]
        }
      >
        Make Transfer
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "position": "absolute",
        "zIndex": -1,
      }
    }
  >
    <RNSVGSvgView
      bbHeight={500}
      bbWidth={750}
      focusable={false}
      height={500}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 500,
            "width": 750,
          },
        ]
      }
      width={750}
    >
      <RNSVGGroup>
        <RNSVGCircle
          cx={450}
          cy={800}
          fill={4278190208}
          propList={
            Array [
              "fill",
              "stroke",
              "strokeWidth",
            ]
          }
          r={500}
          stroke={4278190208}
          strokeWidth="2"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Modal
    animationType="none"
    deviceHeight={null}
    deviceWidth={null}
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    panResponderThreshold={4}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    testID="detailModal"
    transparent={true}
    visible={false}
  />
  <Modal
    animationType="none"
    deviceHeight={null}
    deviceWidth={null}
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    panResponderThreshold={4}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  />
</View>
`;

exports[`Dashboard Page  open modal for detail transaction 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "position": "absolute",
        "top": 0,
      }
    }
  >
    <RNSVGSvgView
      bbHeight={500}
      bbWidth={750}
      focusable={false}
      height={500}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 500,
            "width": 750,
          },
        ]
      }
      width={750}
    >
      <RNSVGGroup>
        <RNSVGCircle
          cx={-50}
          cy={-300}
          fill={4278190208}
          propList={
            Array [
              "fill",
              "stroke",
              "strokeWidth",
            ]
          }
          r={500}
          stroke={4278190208}
          strokeWidth="2"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    collapsable={false}
    style={
      Object {
        "alignSelf": "flex-end",
        "backgroundColor": "#f2f2f2",
        "margin": 10,
        "transform": Array [
          Object {
            "translateX": 55,
          },
        ],
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "gray",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ó°—½
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginLeft": 5,
            "opacity": 1,
          }
        }
      >
        <Text>
          Logout
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
          "backgroundColor": "#fff",
          "borderBottomEndRadius": 50,
          "borderColor": "navy",
          "borderTopEndRadius": 50,
          "borderWidth": 2,
          "elevation": 20,
          "opacity": 0.9,
          "padding": 20,
          "width": 600,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
      testID="money"
    >
      You have
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 30,
            "fontWeight": "bold",
          },
        ]
      }
    >
      SGD$
      <Text
        testID="balance"
      >
        111,111.00
      </Text>
    </Text>
    <Text>
      Account No
    </Text>
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
      testID="accountNo"
    >
      12345
    </Text>
    <Text>
      Account Holder
    </Text>
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 20,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 20,
            "fontWeight": "bold",
          },
          Object {
            "marginBottom": 10,
          },
        ]
      }
    >
      Your transaction history
    </Text>
    <RCTScrollView
      contentContainerStyle={
        Object {
          "paddingBottom": 50,
        }
      }
      data={
        Array [
          Object {
            "amount": 100000,
            "description": "Default money-in transaction",
            "sender": Object {
              "accountHolder": "Jane",
              "accountNo": "1111-111-1111",
            },
            "transactionDate": "2022-03-11T19:48:18.249Z",
            "transactionId": "622ba782f3aa395d71354900",
            "transactionType": "received",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      testID="transaction"
      viewabilityConfigCallbackPairs={Array []}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="detail"
          >
            <View
              style={
                Object {
                  "backgroundColor": "#fff",
                  "borderColor": "#eaeaea",
                  "borderRadius": 20,
                  "borderWidth": 2,
                  "height": 100,
                  "marginVertical": 5,
                  "padding": 10,
                }
              }
            >
              <Text>
                3/12/2022, 2:48:18 AM
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "flexDirection": "row",
                    },
                    Object {
                      "justifyContent": "space-between",
                    },
                  ]
                }
              >
                <View>
                  <Text>
                    Jane
                  </Text>
                  <Text>
                    1111-111-1111
                  </Text>
                </View>
                <View>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#20cb9d",
                        },
                        Object {
                          "fontWeight": "bold",
                        },
                      ]
                    }
                  >
                    +SGD$
                    100,000.00
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 0,
        "left": 0,
        "marginBottom": 20,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "black",
          "borderRadius": 50,
          "opacity": 1,
          "paddingHorizontal": 50,
          "paddingVertical": 10,
        }
      }
      testID="goToTransfer"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontSize": 20,
              "fontWeight": "bold",
            },
            Object {
              "color": "#fff",
            },
          ]
        }
      >
        Make Transfer
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "position": "absolute",
        "zIndex": -1,
      }
    }
  >
    <RNSVGSvgView
      bbHeight={500}
      bbWidth={750}
      focusable={false}
      height={500}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 500,
            "width": 750,
          },
        ]
      }
      width={750}
    >
      <RNSVGGroup>
        <RNSVGCircle
          cx={450}
          cy={800}
          fill={4278190208}
          propList={
            Array [
              "fill",
              "stroke",
              "strokeWidth",
            ]
          }
          r={500}
          stroke={4278190208}
          strokeWidth="2"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Modal
    animationType="none"
    deviceHeight={null}
    deviceWidth={null}
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    panResponderThreshold={4}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    testID="detailModal"
    transparent={true}
    visible={true}
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      forwardedRef={[Function]}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      collapsable={false}
      deviceHeight={null}
      deviceWidth={null}
      forwardedRef={[Function]}
      hideModalContentWhileAnimating={false}
      onBackdropPress={[Function]}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      panResponderThreshold={4}
      pointerEvents="box-none"
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      statusBarTranslucent={false}
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "margin": 37.5,
          "transform": Array [
            Object {
              "translateY": 1334,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "padding": 10,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 20,
                "fontWeight": "bold",
              },
              Object {
                "textAlign": "center",
              },
            ]
          }
        >
          Details Transaction
        </Text>
        <Text />
        <Text>
          Transaction type:
           
          <Text
            style={
              Array [
                Object {
                  "fontWeight": "bold",
                },
                Object {
                  "color": "#000",
                },
              ]
            }
          >
            RECEIVED
          </Text>
        </Text>
        <Text>
          3/12/2022, 2:48:18 AM
           
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            ó°¥”
          </Text>
        </Text>
        <Text />
        <Text
          style={
            Array [
              Object {
                "fontWeight": "bold",
              },
              Object {
                "color": "#000",
              },
            ]
          }
        >
          SENDER
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "width": 100,
              }
            }
          >
            Name:
          </Text>
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            : 
            
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "width": 100,
              }
            }
          >
            Account:
          </Text>
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            : 
            12345
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "width": 100,
              }
            }
          >
            Amount: 
          </Text>
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            : SGD$
            100,000.00
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "width": 100,
              }
            }
          >
            Description: 
          </Text>
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            : 
            Default money-in transaction
          </Text>
        </View>
        <Text />
        <Text
          style={
            Array [
              Object {
                "fontWeight": "bold",
              },
              Object {
                "color": "#000",
              },
            ]
          }
        >
          RECEIPIENT
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "width": 100,
              }
            }
          >
            Name
          </Text>
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            : 
            Jane
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "width": 100,
              }
            }
          >
            Account
          </Text>
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            : 
            1111-111-1111
          </Text>
        </View>
        <Text />
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "width": 100,
              }
            }
          >
            Transaction Id:
          </Text>
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            : 
            622ba782f3aa395d71354900
          </Text>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "flex-end",
              "backgroundColor": "navy",
              "borderRadius": 20,
              "marginTop": 10,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
              }
            }
          >
            Close
          </Text>
        </View>
      </View>
    </View>
  </Modal>
  <Modal
    animationType="none"
    deviceHeight={null}
    deviceWidth={null}
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    panResponderThreshold={4}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  />
</View>
`;
